### PR TITLE
[HFH] More Agentic Acceptance Criteria tweaks

### PR DESCRIPTION
**Pull request by garettarrowood**
## Summary

Refines the agentic acceptance-criteria workflow based on additional QA feedback. Generated plans now prioritize complete page coverage, use UI-facing permission names, explicitly call out permission changes, and present test metadata in a more compact, copy-friendly format.

## What Changed

- Requires baseline coverage for every distinctly affected reachable page before expanding any single area with edge cases.
- Prevents behaviorally different pages from being grouped solely because they share underlying code.
- Uses Consent’s matching action to resolve the UI-facing subject label and titleizes action keys as Nitro does.
- Adds a required `permissions.changes` array for permission definitions and direct authorization lookups added, removed, or modified by the diff.
- Displays permission-change notes prominently in the top Permissions / Roles section.
- Removes backticks from landing pages and permission displays.
- Promotes section, feature-area, and scenario headings by one level.
- Separates scenarios from their metadata while keeping Landing Page and Permissions tightly grouped.
- Displays multiple scenario permissions on one line.
- Adds spacing between Applicable Functional Cases and additional regression verification checks.
- Updates documentation and examples for the revised JSON contract and Markdown output.

## Validation

- [x] Acceptance-criteria parser specs — 13 examples, 0 failures
- [x] Acceptance-criteria formatter specs — 9 examples, 0 failures
- [x] Command-line renderer specs — 2 examples, 0 failures
- [x] All 24 examples pass under `LANG=C` and `LC_ALL=C`
- [x] Ruby syntax checks pass
- [x] Composite-action YAML parses successfully
- [x] `git diff --check` passes

### CODE REVIEW

File: agentic-acceptance-criteria/scripts/acceptance_criteria_formatter.rb [hunk 1]
```diff
@@ -41,11 +41,17 @@ def heading
...
+      "## Permissions / Roles",
       "",
       "- **Required Permissions:** #{PERMISSION_LABELS.fetch(@parsed.permissions.fetch("required"))}",
     ]
 
+    permission_changes = @parsed.permissions.fetch("changes")
+    unless permission_changes.empty?
+      lines << "- **Permission Changes in This PR:**"
+      permission_changes.each { |change| lines << "  - #{change.delete("`")}" }
+    end
```
[the remark this answers is not in the document]
Comment (garettarrowood) [reply agentic-acceptance-criteria/scripts/acceptance_criteria_formatter.rb:53]: Nah - I like it this way.

**Comment by garettarrowood** [comment]
Example of format this generates (and previous) on this PR = https://github.com/powerhome/nitro-web/pull/61538#issuecomment-5183829303

**Review by wadewinningham — APPROVED** [review]
